fuel/streams.py

```diff
@@ -37,11 +37,19 @@ class AbstractDataStream(object):
     sources : tuple of strings
         The names of the data sources returned by this data stream, as
         given by the dataset.
+    produces_examples : bool
+        Whether this data stream produces examples (as opposed to batches
+        of examples).
 
     """
     def __init__(self, iteration_scheme=None, axis_labels=None):
         self.iteration_scheme = iteration_scheme
         self.axis_labels = axis_labels
+        if self.iteration_scheme:
+            self.produces_examples = iteration_scheme.requests_examples
+        else:
+            # TODO: is this a valid assumption?
```

Comment:
So I guess in concrete terms that would just mean setting `self.produces_examples = True` if there's no iteration scheme in the `DataStream` constructor.